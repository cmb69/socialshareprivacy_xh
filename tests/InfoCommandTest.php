<?php

namespace Socialshareprivacy;

use ApprovalTests\Approvals;
use PHPUnit\Framework\TestCase;
use Plib\FakeSystemChecker;
use Plib\SystemChecker;
use Plib\View;

class InfoCommandTest extends TestCase
{
    private SystemChecker $systemChecker;
    private View $view;

    protected function setUp(): void
    {
        global $plugin_tx;
        $plugin_tx = XH_includeVar("./languages/en.php", "plugin_tx");
        $this->systemChecker = new FakeSystemChecker();
        $this->view = new View("./views/", $plugin_tx["socialshareprivacy"]);
    }

    private function sut(): InfoCommand
    {
        return new InfoCommand("./plugins/socialshareprivacy", $this->systemChecker, $this->view);
    }

    public function testRendersSystemCheck(): void
    {
        $response = $this->sut()();
        $this->assertSame("Socialshareprivacy 2.0RC1", $response->title());
        Approvals::verifyHtml($response->output());
    }
}
