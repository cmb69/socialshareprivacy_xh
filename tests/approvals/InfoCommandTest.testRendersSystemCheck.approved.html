
<article class="socialshareprivacy_pluginfo">
  <h1>Socialshareprivacy 2.0-dev</h1>
  <section class="socialshareprivacy_syscheck">
    <h2>System check</h2>
    <p class="xh_fail">PHP version ≥ 7.4.0</p>
    <p class="xh_fail">CMSimple_XH version ≥ 1.7.0</p>
    <p class="xh_fail">Plib_XH version ≥ 1.10</p>
    <p class="xh_warning">Folder './plugins/socialshareprivacyconfig/' writable</p>
    <p class="xh_warning">Folder './plugins/socialshareprivacycss/' writable</p>
    <p class="xh_warning">Folder './plugins/socialshareprivacylanguages/' writable</p>
  </section>
</article>
